### PR TITLE
Profiles bug

### PR DESCRIPTION
**Pull request by Q-Sharp**
Fix bugs in profile command + add new prop. to config
Its possible to show or hide historic user profiles
Every discord profle which is not findable by the bot counts as historic (user left all bl channels)

refactored / code cleanup + add comments

### CODE REVIEW

File: cogs/profiles.py [hunk 1]
```diff
@@ -178,50 +177,47 @@ async def profile(self, ctx, *, userName:str = None):
...
             for guild in self.bot.guilds:
-                members = list(filter(lambda m: foundUserName in m.display_name, guild.members))
-                for member in members:
+                for member in list(filter(lambda m: userName in m.display_name, guild.members)):
                     userids.append(member.id)
 
             for profil in profiles:
-                if foundUserName in profiles[profil]['Base']['username']:
+                if userName in profiles[profil]['Base']['username']:
                   userids.append(int(profil))
```
Comment (Greenfoot5) [inline cogs/profiles.py]: Can `profil` be renamed to `profile` so that it's no longer a typo?

**Review by Greenfoot5 — COMMENTED** [review]
I don't know if you included this, but it's possible to get information on users who the bot can't see. I think it's `bot.fetch_user`.

File: cogs/profiles.py (hunk 1 above)
Comment (Q-Sharp) [reply cogs/profiles.py]: Fixed :)

**Comment by Q-Sharp** [comment]
> 
> 
> I don't know if you included this, but it's possible to get information on users who the bot can't see. I think it's `bot.fetch_user`.

I already use this:

await bot_fetch_user(userid)

**Comment by Q-Sharp** [comment]
Woops. I'll merge this now.

**Pull request merged**